driver/src/orderbook/streamed/state.rs

```diff
@@ -68,7 +68,7 @@ impl State {
                 batch_id
             }
         };
-        Ok((self.account_state(batch_id), self.orders(batch_id)))
+        Ok((self.account_state(batch_id), self.orders(batch_id - 1)))
```

Comment:
Here we make it so that `orderbook_for_batch` returns orders that are valid in the same batch as the balances it returns.

Reply:
I see a problem now:

```
batch 10 (batch to solve 9):
there is a deposit, deposit batch is set to 10
there is a trade of an order, order has pending used amount set to batch 10.
```
We want orders that have to change `fn orders` so that it uses one batch_id for `order_is_valid_in_batch` and that batch_id + 1 for `as_model_order`.

Reply:
So we subtract one from the batch ID here only to add it again in the internal function?